src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPInterfaceProperties.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace System.Net.NetworkInformation
+{
+    internal class LinuxIPInterfaceProperties : UnixIPInterfaceProperties
+    {
+        private readonly LinuxNetworkInterface _linuxNetworkInterface;
+        private readonly GatewayIPAddressInformationCollection _gatewayAddresses;
+        private readonly IPAddressCollection _dhcpServerAddresses;
+        private readonly IPAddressCollection _winsServerAddresses;
+        private readonly LinuxIPv4InterfaceProperties _ipv4Properties;
+        private readonly LinuxIPv6InterfaceProperties _ipv6Properties;
+
+        public LinuxIPInterfaceProperties(LinuxNetworkInterface lni)
+            : base(lni)
+        {
+            _linuxNetworkInterface = lni;
+            _gatewayAddresses = GetGatewayAddresses();
+            _dhcpServerAddresses = GetDhcpServerAddresses();
+            _winsServerAddresses = GetWinsServerAddresses();
+            _ipv4Properties = new LinuxIPv4InterfaceProperties(lni);
+            _ipv6Properties = new LinuxIPv6InterfaceProperties(lni);
+        }
+
+        public override bool IsDynamicDnsEnabled
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override IPAddressInformationCollection AnycastAddresses
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override GatewayIPAddressInformationCollection GatewayAddresses { get { return _gatewayAddresses; } }
+
+        public override IPAddressCollection DhcpServerAddresses
+        {
+            get
+            {
+                return _dhcpServerAddresses;
+            }
+        }
+
+        public override IPAddressCollection WinsServersAddresses
+        {
+            get
+            {
+                return _winsServerAddresses;
+            }
+        }
+
+        public override IPv4InterfaceProperties GetIPv4Properties()
+        {
+            return _ipv4Properties;
+        }
+
+        public override IPv6InterfaceProperties GetIPv6Properties()
+        {
+            return _ipv6Properties;
+        }
+
+        // /proc/net/route contains some information about gateway addresses,
+        // and seperates the information about by each interface.
+        public GatewayIPAddressInformationCollection GetGatewayAddresses()
+        {
+            GatewayIPAddressInformationCollection collection = new GatewayIPAddressInformationCollection();
+            // Columns are as follows (first-line header):
+            // Iface  Destination  Gateway  Flags  RefCnt  Use  Metric  Mask  MTU  Window  IRTT
```

Comment:
We should ensure this assumption regarding the header is true (in other places as well) through either an integration test (PAL or FunctionalTest) or through a `DEBUG` only check.